### PR TITLE
Resize support

### PR DESCRIPTION
**Pull request by pkalever**
Earlier these changes where tracked/reviewed at:
https://review.gluster.org/#/c/18787/
https://review.gluster.org/#/c/19665/

Please refer the links for complete context.

### CODE REVIEW

**Comment by pkalever** [comment]
please-review: @pranithk @lxbsz @vbellur

File: rpc/block_svc_routines.c [hunk 1]
```diff
@@ -2774,6 +2946,265 @@ block_modify_cli_1_svc_st(blockModifyCli *blk, struct svc_req *rqstp)
...
+
+  asyncret = glusterBlockModifySizeRemoteAsync(info, glfs, &mobj, &savereply);
+  if (asyncret) {   /* asyncret decides result is success/fail */
+    errCode = asyncret;
+    LOG("mgmt", GB_LOG_WARNING,
+        "glusterBlockModifySizeRemoteAsync(size=%zu): return %d %s for block %s on volume %s",
+        blk->size, asyncret, FAILED_REMOTE_AYNC_MODIFY, blk->block_name, info->volume);
+    goto out;
+  } else {
+    GB_METAUPDATE_OR_GOTO(lock, glfs, mobj.block_name, mobj.volume,
```
Comment (pkalever) [inline rpc/block_svc_routines.c]: Not sure, if we want to do this in else block only.

File: rpc/block_svc_routines.c [hunk 2]
```diff
@@ -1450,6 +1611,17 @@ glusterBlockBuildMinCaps(void *data, operations opt)
       minCaps[GB_JSON_CAP] = true;
     }
     break;
+  case MODIFY_SIZE_SRV:
+    msblk = (blockModifySizeCli *)data;
+
+    minCaps[GB_MODIFY_CAP] = true;
```
Comment (pranithk) [inline rpc/block_svc_routines.c]: We can remove the line above, considering there is no relation between these two capabilities now.

File: rpc/glfs-operations.c [hunk 3]
```diff
@@ -233,6 +233,88 @@ glusterBlockCreateEntry(struct glfs *glfs, blockCreateCli *blk, char *gbid,
...
+      *errCode = errno;
+      LOG("gfapi", GB_LOG_ERROR,
+          "glfs_ftruncate(%s): on volume %s for block %s "
+          "of size %zu failed[%s]", blk->gbid, blk->volume, blk->block_name,
+          blk->size, strerror(errno));
+      goto close;
+    }
+
+    /* dirty hack to check if the file is zerofilled ? */
+    if ((blk->size - sb.st_size > 0) && (sb.st_size <= 512 * sb.st_blocks)) {
```
Comment (pranithk) [inline rpc/glfs-operations.c]: Could you change this to if ((blk->size > sb.st_size))

File: rpc/block_svc_routines.c (hunk 1 above)
Comment (pranithk) [reply rpc/block_svc_routines.c]: no problem I think

**Comment by pkalever** [comment]
@pranithk updated the patches with the review comments.

**Comment by pkalever** [comment]
@pranithk tested and updated.

**Pull request merged**